### ISSUE TITLE
Remove `<!--#include` from viewer.html

### ISSUE THREAD

**Issue by yurydelendik** [mozilla/pdf.js#7725]
To increase maintainability of the viewer.html we probably need to remove `<!--#include`. We have many tiny viewer-snippet-*.html files. There are only present because we don't want some markup html code present in developer viewer.html. 

We also have issue with comments inside `<!--#if`. Some thoughts on solving it is to hide dom element by enclosing them in `<script>` tag, e.g.

    
    <script type="pdfjs-preprocessor">
        <div>Something hidden from dev viewer</div>
        <x-script src="somescript.js"></x-script>
    </script>
    

Shall we even remove every `<!--#` and replace then with dom friendly, e.g. `<div pdfjs-if="PRODUCTION">` and `<script type="pdfjs-preprocessor" pdfjs-if="PRODUCTION">...` ?

**Comment by timvandermeij** [comment]
I personally think we should get rid of the `<!--#` comments altogether, so in that case your last solution would probably be best.

**Comment by himanish-star** [comment]
@timvandermeij , can I work on this issue?

**Comment by timvandermeij** [comment]
Yes, but I'd recommend to first talk with e.g. Yury on IRC about what a good approach to fix this would be.

**Comment by himanish-star** [comment]
ok @timvandermeij , I'll surely do this. Thanks.

**Comment by yurydelendik** [comment]
My vision for this bug is too construct entire viewer HTML using JS. We can do it in pieces. So at the end entire viewer.html will be almost empty, but components (such as toolbar,  app, password prompt) will render HTML by them self.

**Comment by himanish-star** [comment]
@yurydelendik , if you could guide me on how to start on this , then I will start this as soon as possible.

**Comment by himanish-star** [comment]
> (assuming there is a preprocessor that can handle this).

@timvandermeij , Could you help me find where the code for this pre-processor is inside `pdf.js`

**Comment by Rob--W** [comment]
Our HTML preprocessor is implemented at
https://github.com/mozilla/pdf.js/blob/55e3f97aa97ad648ef955203524bbdb5593671d1/external/builder/builder.js

The constants are defined at

https://github.com/mozilla/pdf.js/blob/55e3f97aa97ad648ef955203524bbdb5593671d1/gulpfile.js#L74-L86

and overridden by calls to `builder.merge`,  e.g. at

https://github.com/mozilla/pdf.js/blob/55e3f97aa97ad648ef955203524bbdb5593671d1/gulpfile.js#L588

and the preprocessor is called by `preprocessHTML`, implemented at:

https://github.com/mozilla/pdf.js/blob/55e3f97aa97ad648ef955203524bbdb5593671d1/gulpfile.js#L560-L568

**Comment by himanish-star** [comment]
@Rob--W, Thanks for the explanation and sorry for the late reply

**Comment by rhunoja** [comment]
I want to use code from https://github.com/mozilla/pdf.js in order use it on my asp webpage, but it doesn't work, there is a js message "PDFJSDev is not defined" i don't know where set it up to pass conditions in viewer.js like..

if (typeof PDFJSDev !== 'undefined' && PDFJSDev.test('CHROME')) {
  var defaultUrl; // eslint-disable-line no-var

Several page have this condition, any idea of what i'm doing wrong?

**Comment by timvandermeij** [comment]
@rhunoja Please don't post unrelated comments on issues. This question has nothing to do with the removal of include statements from the viewer.
